presto-spark-base/src/main/java/com/facebook/presto/spark/execution/PrestoSparkTaskExecutorFactory.java

```diff
@@ -276,59 +281,74 @@ public IPrestoSparkTaskExecutor doCreate(
                 allocationTrackingEnabled,
                 false);
 
-        OutputBufferMemoryManager memoryManager = new OutputBufferMemoryManager(
-                sinkMaxBufferSize.toBytes(),
-                () -> queryContext.getTaskContextByTaskId(taskId).localSystemMemoryContext(),
-                notificationExecutor);
-        PrestoSparkRowBuffer rowBuffer = new PrestoSparkRowBuffer(memoryManager);
-
-        ImmutableMap.Builder<PlanNodeId, Iterator<PrestoSparkMutableRow>> shuffleInputs = ImmutableMap.builder();
-        ImmutableMap.Builder<PlanNodeId, Iterator<PrestoSparkSerializedPage>> broadcastInputs = ImmutableMap.builder();
+        ImmutableMap.Builder<PlanNodeId, Iterator<PrestoSparkMutableRow>> rowInputs = ImmutableMap.builder();
+        ImmutableMap.Builder<PlanNodeId, Iterator<PrestoSparkSerializedPage>> pageInputs = ImmutableMap.builder();
         for (RemoteSourceNode remoteSource : fragment.getRemoteSourceNodes()) {
-            List<Iterator<PrestoSparkMutableRow>> shuffleRemoteSourceInputs = new ArrayList<>();
-            List<Iterator<PrestoSparkSerializedPage>> broadcastRemoteSourceInputs = new ArrayList<>();
+            List<Iterator<PrestoSparkMutableRow>> remoteSourceRowInputs = new ArrayList<>();
+            List<Iterator<PrestoSparkSerializedPage>> remoteSourcePageInputs = new ArrayList<>();
             for (PlanFragmentId sourceFragmentId : remoteSource.getSourceFragmentIds()) {
                 Iterator<Tuple2<MutablePartitionId, PrestoSparkMutableRow>> shuffleInput = inputs.getShuffleInputs().get(sourceFragmentId.toString());
                 Broadcast<List<PrestoSparkSerializedPage>> broadcastInput = inputs.getBroadcastInputs().get(sourceFragmentId.toString());
-                checkArgument(shuffleInput != null || broadcastInput != null, "Input not found for sourceFragmentId: %s", sourceFragmentId);
-                checkArgument(shuffleInput == null || broadcastInput == null, "Single remote source cannot accept both, broadcast and shuffle inputs");
+                List<PrestoSparkSerializedPage> localInput = inputs.getLocalInputs().get(sourceFragmentId.toString());
```

Comment:
what's local input? Input from "VALUES" node? -- maybe add a comment to it.

Reply:
Those are "inmemory" inputs. For the COORDINATOR_ONLY fragment we first collect the inputs on the Driver, and provide them to the fragment.

Reply:
@arhimondr : Add comment about this "localInputs" some where then.

Reply:
Going to add a comment in `PrestoSparkTaskInputs`